src/modules/beamos/filesystem/migration/usr/bin/migrationos_to_beamos2.sh

```diff
@@ -49,37 +71,73 @@ PHASE_COLOR="ORANGE"
 # Set trap before EXIT
 trap do_exit EXIT
 
+# find the right device name of partition 3 on the USB stick and mount it
+echo "$(timestamp) $0: Mounting the USB stick partition 3"
+USB_PART=$(lsblk -o NAME,SIZE -d -p -n | grep "/dev/sd" | awk '{print $1}')
+USB_PARTITION="${USB_PART}3"
+sudo mkdir -p ${LOG_LOCATION}
+sudo fsck -a ${USB_PARTITION}
+sudo mount ${USB_PARTITION} ${LOG_LOCATION}
+exit_code=$?
+if [ "$exit_code" -ne 0 ]; then
+  echo "$(timestamp) $0: mount ${USB_PARTITION} on ${LOG_LOCATION} failed"
+fi
+echo "$(timestamp) $0: Mounting the USB stick partition 3 successful"
+
 # start progress leds
 current_pid=$$
-sudo bash ${BASEDIR}/progress.sh $current_pid $MAX_TIME $PHASE_COLOR &
+sudo bash ${BASEDIR}/progress.sh $current_pid $MAX_TIME $PHASE_COLOR &>> $LOG_FILE &
 
 #   Flash the SD-Card
-echo "$(timestamp) $0: Flashing the SD-Card"
-sudo bash ${BASEDIR}/migration.sh flash beamos2 sd-card
+log "$(timestamp) $0: Flashing the SD-Card"
+cmd 'sudo bash ${BASEDIR}/migration.sh flash beamos2 sd-card'
 exit_code=$?
 if [ "$exit_code" -ne 0 ]; then
-  echo "$(timestamp) $0: Flashing the SD-Card failed"
+  log "$(timestamp) $0: Flashing the SD-Card failed"
   exit 100
 fi
 
 #   Mount the SD-Card
-echo "$(timestamp) $0: Mounting the SD-Card"
-sudo bash ${BASEDIR}/migration.sh mount sd-card
+log "$(timestamp) $0: Mounting the SD-Card"
+cmd 'sudo bash ${BASEDIR}/migration.sh mount sd-card'
 exit_code=$?
 if [ "$exit_code" -ne 0 ]; then
-  echo "$(timestamp) $0: Mounting the SD-Card failed"
+  log "$(timestamp) $0: Mounting the SD-Card failed"
   exit 100
 fi
 
 #   Restore Sensitive Data
-echo "$(timestamp) $0: Restoring Sensitive Data"
-sudo bash ${BASEDIR}/migration.sh restore-data
+log "$(timestamp) $0: Restoring Sensitive Data"
+cmd 'sudo bash ${BASEDIR}/migration.sh restore-data'
 exit_code=$?
 if [ "$exit_code" -ne 0 ]; then
-  echo "$(timestamp) $0: Restoring Sensitive Data failed"
+  log "$(timestamp) $0: Restoring Sensitive Data failed"
   exit 100
 fi
 
+# Copy syslog to the USB stick
+log "$(timestamp) $0: Copying syslog to the USB stick"
+# Check if the file exists and rename it if it does
+if [ -f ${SYSLOG_BACKUP} ]; then
+  cmd 'sudo mv ${SYSLOG_BACKUP} ${SYSLOG_BACKUP}.old'
+fi
+cmd 'sudo cp ${SYSLOG_LOCATION} ${SYSLOG_BACKUP}'
+exit_code=$?
+if [ "$exit_code" -ne 0 ]; then
+  log "$(timestamp) $0: Copy syslog to the USB stick failed"
+fi
+log "$(timestamp) $0: Copying syslog to the USB stick successful"
+
+# Copy logs on the /mnt/usb to the freshly flashed SD-Card rootfs var log partition
+log "$(timestamp) $0: Copying logs to the freshly flashed SD-Card rootfs var log partition"
+log "$(timestamp) $0: System will reboot after this........"
+sudo cp -r ${LOG_LOCATION}/* ${SDCARD_ROOTFS_A_PATH}/var/log/
```

Comment:
Why not we use `cmd` then the command here?

Reply:
At this stage, logs are already copied and ready to reboot.
So using cmd or log functions would not make a difference to the backed up logs